runtime/lua/vim/lsp.lua

```diff
@@ -577,21 +577,32 @@ function lsp.enable(name, enable)
   end
 
   if not next(lsp._enabled_configs) then
+    -- If there are no remaining LSPs enabled, remove the enable autocmd.
     if lsp_enable_autocmd_id then
       api.nvim_del_autocmd(lsp_enable_autocmd_id)
       lsp_enable_autocmd_id = nil
     end
-    return
+  else
+    -- Only ever create autocmd once to reuse computation of config merging.
+    lsp_enable_autocmd_id = lsp_enable_autocmd_id
+      or api.nvim_create_autocmd('FileType', {
```

Comment:
normally in favor of ternaries but in this case, is this clearer?

```suggestion
  elseif not lsp_enable_autocmd_id then
    -- Only ever create autocmd once to reuse computation of config merging.
    lsp_enable_autocmd_id = api.nvim_create_autocmd('FileType', {
```

Reply:
I agree the ternary is dense. I also didn't write it, and it looks like you've merged the PR :)